lightningd/pay.c

```diff
@@ -444,7 +444,7 @@ static void json_pay(struct command *cmd,
 
 	if (b11->msatoshi) {
 		msatoshi = *b11->msatoshi;
-		if (msatoshitok) {
+		if (msatoshitok && !json_tok_is_null(buffer, msatoshitok)) {
```

Comment:
My reading of `json_get_params` suggests that a null input sets corresponding `jsmntok` to `NULL`, maybe unnecessary?

Reply:
Oh, it does too!  Clever old Rusty, dumb modern Rusty.